Bottom: 1c84a3ca61dc6d7cb32d51325473049ff7b3746c
Top:    a888a0025c245c66b3c227a1b877634fef6bbfed
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-24 00:48:40 +0100

Attempt to debug islandness after inserting hole bug


---

diff --git a/src/action.c b/src/action.c
index 6121665..2f306f8 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1507,12 +1507,19 @@ NotifyMode (void)
 		   */
 		  SaveUndoSerialNumber ();
 		  Flags = ((PolygonType *)Crosshair.AttachedObject.Ptr2)->Flags;
-		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
-				      result, Flags);
+#warning FIXME: Removing a pour should force a re-check of islanding!
+                  /* For now, do this before PolyToPours() which re-inits the
+                   * clipping for the pour(s) affected (?). Ideally RemoveObject
+                   * should fix any island response its self. Anyway.. removing
+                   * the old thing first saves us doing twice the work by
+                   * recomputing islands when it gets deleted.
+                   */
 		  RemoveObject (POUR_TYPE,
 				Crosshair.AttachedObject.Ptr1,
 				Crosshair.AttachedObject.Ptr2,
 				Crosshair.AttachedObject.Ptr3);
+		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
+				      result, Flags);
 		  RestoreUndoSerialNumber ();
 		  IncrementUndoSerialNumber ();
 		  Draw ();
